src/test/pybind/test_ceph_argparse.py

```diff
@@ -1066,11 +1066,8 @@ def test_pool_create(self):
         self.assert_valid_command(['osd', 'pool', 'create',
                                    'poolname', '128', '128',
                                    'erasure', 'A-Za-z0-9-_.', 'ruleset^^'])
+        self.assert_valid_command(['osd', 'pool', 'create', 'poolname'])
         assert_equal({}, validate_command(sigdict, ['osd', 'pool', 'create']))
-        assert_equal({}, validate_command(sigdict, ['osd', 'pool', 'create',
-                                                    'poolname']))
-        assert_equal({}, validate_command(sigdict, ['osd', 'pool', 'create',
```

Comment:
i get your point. but the commit message is not quite right.

> note that we remove the bad pg_num = -1 test too, which will now be treated as the the crush rule name (invalid but allowed)

crush rule named "-1" is indeed unusual, but it's not invalid. neither crush rule parser nor monitor disallows it.

i'd suggest take this opportunity to fix the command description of "osd pool create", like

```diff
diff --git a/src/mon/MonCommands.h b/src/mon/MonCommands.h
index 273b019914..e188f197d2 100644
--- a/src/mon/MonCommands.h
+++ b/src/mon/MonCommands.h
@@ -1022,8 +1022,8 @@ COMMAND("osd pool create " \
         "name=pool_type,type=CephChoices,strings=replicated|erasure,req=false " \
    "name=erasure_code_profile,type=CephString,req=false,goodchars=[A-Za-z0-9-_.] " \
    "name=rule,type=CephString,req=false " \
-        "name=expected_num_objects,type=CephInt,req=false " \
-        "name=size,type=CephInt,req=false " \
+        "name=expected_num_objects,type=CephInt,range=0,req=false " \
+        "name=size,type=CephInt,range=0,req=false " \
    "name=pg_num_min,type=CephInt,range=0,req=false " \
    "name=target_size_bytes,type=CephInt,range=0,req=false " \
    "name=target_size_ratio,type=CephFloat,range=0|1,req=false",\
``` 
in a separated commit. and replace this test with another one, to ensure that our coverage don't shrink over time, also to make sure we can catch at least *some* invalid command sequences even if this is not completely foolproof. for instance:
```
        assert_equal({}, validate_command(sigdict, ['osd', 'pool', 'create',
                                                    'poolname',
                                                    '-1', '-1',
                                                    'ruleset']))
```
this might not be the best test, but it shows that the parser is able to identify some of the invalid cases. as the invalid pg num and pgp num will spill over to other arguments which are more tolerant, the following argument won't be matched as the command description of certain argument(s) have been matched.

Reply:
#30538

Reply:
back from a meeting just now :-(